Automattic/wp-calypso
# ETK: Fix for dotcom legacy FSE to work with current Gutenberg versions

#### Changes proposed in this Pull Request

* Update Dotcom legacy FSE to be compatible with the current version of Gutenberg

#### Testing instructions

For ease of testing, I've created D63129-code with these changes along with updating which version of Gutenberg is loaded, for testing on Simple Sites.

**Test block migrations when upgrading Gutenberg**

- Create a Simple site that has legacy FSE enabled
  - Normally this can be done by following the onboarding flow from https://horizon.wordpress.com/start/test-fse/fse-themes. A recent regression prevents us from doing so.
  - The alternative is creating a simple wordpress site, updating a legacy FSE compatible theme, and updating the blog sticker for said simple site.
    - Create a simple site https://wordpress.com/start/domains
    - Open an ssh connection to your dotcom sandbox in a terminal
    - Run this command in the terminal `wp blog-stickers add --who=YOURUSERNAME --url=YOURTESTSITEDOMAIN --sticker=full-site-editing`
    - Run this command in the terminal `wp theme activate "pub/maywood" --url=YOURTESTSITEDOMAIN`
    - Navigate to any page to verify that legacy FSE is enabled
- BEFORE you sandbox the site, import the file in D63129-code that has a post with all the [Core blocks available in Gutenberg v9.7.4](https://github.com/WordPress/gutenberg/tree/release/9.7/packages/block-library/src) 
- Check that the post imported successfully to your site
- Apply D63129-code, sandbox the site, and reload the post in the editor
- Notice any errors in the JS console that may be related to block migrations _for Core blocks_
- Save the post and make sure all blocks display as expected on the front-end

**Test Compatibility of Legacy FSE Features**

- Create a Simple site that has legacy FSE enabled, if you don't have one already (see above for instructions)
- Apply D63129-code in your sandbox, and sandbox both the site domain _AND_ public-api.wordpress.com
- Try editing a page, including changing the header and footer
- Check that the changes save correctly and can be viewed on the front-end of the site

**Test on Atomic Sites**

- The easiest way to test on Atomic sites is the create a Simple Site using the above instructions, purchase a Business plan, and then install a plugin to transfer the site to Atomic. Note that [Legacy FSE features are currently broken on initial transfer](https://github.com/Automattic/wp-calypso/issues/41825)--you'll need to switch the theme, then switch back to Maywood or another legacy FSE compatible theme to scaffold the template content on theme activation.
- Next download a build of the plugin from this branch, and upload to your site for testing, as described in PCYsg-ly5-p2
- Check that you can edit template parts, save the changes, and they show up correctly on the site.

### Deploy note

To deploy this to wpcom, we'll need to combine the ETK changes from this PR with the changes to Gutenberg loading in D63129-code.

Atomic sites will just need the usual ETK release.
